gradient_utils/metrics.py

```diff
@@ -53,6 +54,49 @@ def get_workload_id():
         return os.getenv(WORKLOAD_ID_ENV)
     return _get_experiment_id()
 
+def add_metrics(metrics, workload_id=None, registry=REGISTRY, push_gateway=None, timeout=30):
+    metrics_logger = MetricsLogger(
+        workload_id=workload_id, registry=registry, push_gateway=push_gateway
+    )
+
+    metrics = [Metric(key, value) for key, value in metrics.items()]
+    for metric in metrics:
+        metrics_logger.add_gauge(metric.key)
+        metrics_logger[metric.key].set(metric.value)
+
+    metrics_logger.push_metrics(timeout)
+
+class Metric:
+    def __init__(self, key, value):
+        # TODO: Is there a way get the setters to do the initialization checks?
+        if not isinstance(key, str):
+            raise ValueError('Key of a metric can only be a string')
+        if not isinstance(value, Number):
+            raise ValueError('Value of a metric can only be a number')
+        self.key = key
+        self.value = value
```

Comment:
Do these need to be `_key` and `_value`? As-is I'm surprised they don't hit the setters.

Reply:
I guess not but it seems like it does hit the setters 
This seems to confirm expected behavior
https://stackoverflow.com/questions/4555932/public-or-private-attribute-in-python-what-is-the-best-way

Reply:
I can add unit tests for this class too.